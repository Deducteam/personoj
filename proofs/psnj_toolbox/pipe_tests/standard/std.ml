let () =
  let out_files =
    [ "simple_imp.lp"; "modus_ponens.lp"; "fa_hyp.lp"; "fa_instantiate.lp" ]
  in
  let check = false in
  (* It fails on fa_hyp_use because lambdapi doesn't handle yet quantifiers *)
  Testing.run_pipe ~check ~out_files ~src:"std.log" Sys.argv.(1)
