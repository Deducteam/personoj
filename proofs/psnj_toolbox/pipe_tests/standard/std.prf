(std
 (simple_imp 0
  (simple_imp-1 nil 3848649362 ("" (flatten) nil nil) nil shostak))
 (modus_ponens 0
  (modus_ponens-1 nil 3848649278
   ("" (flatten)
    (("" (split) (("1" (propax) nil nil) ("2" (propax) nil nil)) nil))
    nil)
   nil shostak))
 (fa_hyp 0
  (fa_hyp-1 nil 3848649119 ("" (flatten) nil nil) nil shostak))
 (fa_instantiate 0
  (fa_instantiate-1 nil 3848650783
   ("" (flatten)
    (("" (instantiate - "zero")
      (("" (split) (("1" (propax) nil nil) ("2" (propax) nil nil))
        nil))
      nil))
    nil)
   ((nat nonempty-type-decl nil std nil)
    (zero const-decl "nat" std nil))
   shostak)))

