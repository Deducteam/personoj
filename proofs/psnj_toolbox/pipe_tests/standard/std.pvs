std : THEORY
BEGIN
  P, Q: bool
  simple_imp : LEMMA P IMPLIES P
  modus_ponens : LEMMA (P IMPLIES Q) IMPLIES P IMPLIES Q
  nat : TYPE+
  zero : nat
  even : [nat -> bool]
  fa_hyp : LEMMA (FORALL (n: nat): P IMPLIES (even(n))) IMPLIES P IMPLIES P
  fa_instantiate : LEMMA (FORALL (n: nat) : P IMPLIES even(n)) IMPLIES P IMPLIES even(zero)
END std
