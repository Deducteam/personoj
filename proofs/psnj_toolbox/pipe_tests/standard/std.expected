=> simple_imp.lp:
require open qfo.encoding.lhol qfo.encoding.propositional_connectives;
require qfo.spec.main;
symbol simple_imp!1 : @Prf (@imp (disj (not qfo.spec.main.P) qfo.spec.main.P) (imp qfo.spec.main.P qfo.spec.main.P)) ≔
begin
  why3;
end;
symbol simple_imp!2 : @Prf (@imp (disj (not qfo.spec.main.P) qfo.spec.main.P) (imp qfo.spec.main.P qfo.spec.main.P)) ≔
begin
  why3;
end;
symbol simple_imp!3 : @Prf (disj (not qfo.spec.main.P) qfo.spec.main.P) ≔
begin
  why3;
end;
symbol simple_imp!4 : @Prf (@imp (disj (not qfo.spec.main.P) qfo.spec.main.P) (imp qfo.spec.main.P qfo.spec.main.P)) ≔
begin
  why3;
end;
=> modus_ponens.lp:
require open qfo.encoding.lhol qfo.encoding.propositional_connectives;
require qfo.spec.main;
symbol modus_ponens!5 : @Prf (@imp (disj (not (imp qfo.spec.main.P qfo.spec.main.Q)) (disj (not qfo.spec.main.P) qfo.spec.main.Q)) (@imp (disj (not (imp qfo.spec.main.P qfo.spec.main.Q)) (disj (not qfo.spec.main.P) qfo.spec.main.Q)) (imp (imp qfo.spec.main.P qfo.spec.main.Q) (imp qfo.spec.main.P qfo.spec.main.Q)))) ≔
begin
  why3;
end;
symbol modus_ponens!6 : @Prf (@imp (disj (not (imp qfo.spec.main.P qfo.spec.main.Q)) (disj (not qfo.spec.main.P) qfo.spec.main.Q)) (@imp (disj (not (imp qfo.spec.main.P qfo.spec.main.Q)) (disj (not qfo.spec.main.P) qfo.spec.main.Q)) (imp (imp qfo.spec.main.P qfo.spec.main.Q) (imp qfo.spec.main.P qfo.spec.main.Q)))) ≔
begin
  why3;
end;
symbol modus_ponens!7 : @Prf (@imp (disj qfo.spec.main.P (disj (not qfo.spec.main.P) qfo.spec.main.Q)) (@imp (disj (not qfo.spec.main.Q) (disj (not qfo.spec.main.P) qfo.spec.main.Q)) (disj (not (imp qfo.spec.main.P qfo.spec.main.Q)) (disj (not qfo.spec.main.P) qfo.spec.main.Q)))) ≔
begin
  why3;
end;
symbol modus_ponens!8 : @Prf (@imp (disj qfo.spec.main.P (disj (not qfo.spec.main.P) qfo.spec.main.Q)) (@imp (disj (not qfo.spec.main.Q) (disj (not qfo.spec.main.P) qfo.spec.main.Q)) (disj (not (imp qfo.spec.main.P qfo.spec.main.Q)) (disj (not qfo.spec.main.P) qfo.spec.main.Q)))) ≔
begin
  why3;
end;
symbol modus_ponens!9 : @Prf (disj (not qfo.spec.main.Q) (disj (not qfo.spec.main.P) qfo.spec.main.Q)) ≔
begin
  why3;
end;
symbol modus_ponens!10 : @Prf (disj qfo.spec.main.P (disj (not qfo.spec.main.P) qfo.spec.main.Q)) ≔
begin
  why3;
end;
=> fa_hyp.lp:
require open qfo.encoding.lhol qfo.encoding.propositional_connectives;
require qfo.spec.main;
symbol fa_hyp!11 : @Prf (@imp (disj (not (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n)))) (disj (not qfo.spec.main.P) qfo.spec.main.P)) (imp (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n))) (imp qfo.spec.main.P qfo.spec.main.P))) ≔
begin
  why3;
end;
symbol fa_hyp!12 : @Prf (@imp (disj (not (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n)))) (disj (not qfo.spec.main.P) qfo.spec.main.P)) (imp (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n))) (imp qfo.spec.main.P qfo.spec.main.P))) ≔
begin
  why3;
end;
symbol fa_hyp!13 : @Prf (disj (not (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n)))) (disj (not qfo.spec.main.P) qfo.spec.main.P)) ≔
begin
  why3;
end;
symbol fa_hyp!14 : @Prf (@imp (disj (not (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n)))) (disj (not qfo.spec.main.P) qfo.spec.main.P)) (imp (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n))) (imp qfo.spec.main.P qfo.spec.main.P))) ≔
begin
  why3;
end;
=> fa_instantiate.lp:
require open qfo.encoding.lhol qfo.encoding.propositional_connectives;
require qfo.spec.main;
symbol fa_instantiate!15 : @Prf (@imp (disj (not (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n)))) (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero))) (@imp (disj (not (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n)))) (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero))) (imp (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n))) (imp qfo.spec.main.P (qfo.spec.main.even qfo.spec.main.zero))))) ≔
begin
  why3;
end;
symbol fa_instantiate!16 : @Prf (@imp (disj (not (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n)))) (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero))) (@imp (disj (not (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n)))) (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero))) (imp (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n))) (imp qfo.spec.main.P (qfo.spec.main.even qfo.spec.main.zero))))) ≔
begin
  why3;
end;
symbol fa_instantiate!17 : @Prf (@imp (disj (not (imp qfo.spec.main.P (qfo.spec.main.even qfo.spec.main.zero))) (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero))) (disj (not (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n)))) (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero)))) ≔
begin
  why3;
end;
symbol fa_instantiate!18 : @Prf (@imp (disj (not (imp qfo.spec.main.P (qfo.spec.main.even qfo.spec.main.zero))) (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero))) (disj (not (all qfo.spec.main.nat (λ (n : El qfo.spec.main.nat), imp qfo.spec.main.P (qfo.spec.main.even n)))) (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero)))) ≔
begin
  why3;
end;
symbol fa_instantiate!19 : @Prf (@imp (disj (not (qfo.spec.main.even qfo.spec.main.zero)) (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero))) (@imp (disj qfo.spec.main.P (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero))) (disj (not (imp qfo.spec.main.P (qfo.spec.main.even qfo.spec.main.zero))) (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero))))) ≔
begin
  why3;
end;
symbol fa_instantiate!20 : @Prf (disj (not (qfo.spec.main.even qfo.spec.main.zero)) (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero))) ≔
begin
  why3;
end;
symbol fa_instantiate!21 : @Prf (disj qfo.spec.main.P (disj (not qfo.spec.main.P) (qfo.spec.main.even qfo.spec.main.zero))) ≔
begin
  why3;
end;
